src/api/groups/groups.ts

```diff
@@ -1,59 +1,84 @@
 import { Router } from "express";
 import { ValidatedRequest, createValidator } from "express-joi-validation";
 import { GroupRequestSchema, GroupSchema } from "../../schemas/group";
+import { unknownErrorLoggerMiddleware } from "../../middlewares/errorHandler";
 import GroupService from "../../services/groups/groupService";
-import { DATA_IS_NOT_ANY_MATCH, GROUP_DOES_NOT_EXIST } from "../../constants/errorConstants";
+import ApiError from "../../errors/ApiError";
 
 const router = Router();
 const service = new GroupService();
 const validator = createValidator();
 
-router.get('/', async (req, res) => {
-    const groups = await service.getGroups();
+router.get('/', async (req, res, next) => {
+    try {
+        const groups = await service.getGroups();
 
-    if (!groups.length) {
-        res.status(404).send(DATA_IS_NOT_ANY_MATCH);
-    }
+        if (!groups.length) {
+            next(ApiError.dataIsNotAnyMatch());
+            return;
+        }
 
-    res.send(groups);
+        res.send(groups);
+    } catch (error: any) {
+        unknownErrorLoggerMiddleware(error, req, res, next);
```

Comment:
middlewares should be used as middleware. To run the next middleware in the chain you have to call next method

Reply:
Thanks, fixed into 03bbdce80f275f0e8c7d776ab1e64d14cdf68ade commit